src-electron/db/query-feature.js

```diff
@@ -24,53 +24,612 @@ const dbApi = require('./db-api.js')
 const dbMapping = require('./db-mapping.js')
 
 /**
- * Get all device type features associated with a list of device type refs
+ * Get all device type features associated with a list of device type refs and an endpoint.
+ * Join ENDPOINT_TYPE_ATTRIBUTE and ATTRIBUTE table to get featureMap attribute associated with the feature,
+ * so the frontend could get and set featureMap bit easier.
+ * Only return features with cluster on the side specified in the deivce type.
+ *
+ * @export
  * @param {*} db
  * @param {*} deviceTypeRefs
+ * @param {*} endpointTypeRef
  * @returns All feature information and device type conformance
- * with associated device type and cluster details
+ * with associated device type, cluster, and featureMap attribute details
  */
-async function getFeaturesByDeviceTypeRefs(db, deviceTypeRefs) {
+async function getFeaturesByDeviceTypeRefs(
+  db,
+  deviceTypeRefs,
+  endpointTypeRef
```

Comment:
I don't understand why "get feature by device type" needs an "endpoint type" all of suddenly?

Reply:
ENDPOINT_TYPE_ATTRIBUTE and ATTRIBUTE table is joined to get info on:
1. featureMap value related to the feature, deciding if the feature is enabled
2. attribute id of featureMap, which is needed when updating the featureMap after toggling a feature